src/Supportive/Config/DeptracConfig.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\Supportive\Config;
+
+use Qossmic\Deptrac\Supportive\DependencyInjection\EmitterType;
+use Symfony\Component\Config\Builder\ConfigBuilderInterface;
+use Symfony\Component\Yaml\Yaml;
+
+final class DeptracConfig implements ConfigBuilderInterface
+{
+    private bool $ignoreUncoveredInternalClasses = true;
+    private bool $useRelativePathFromDepfile = true;
+
+    /** @var array<string> */
+    private array $paths = [];
+    /** @var array<LayerConfig> */
+    private array $layers = [];
+    /** @var array<FormatterConfig> */
+    private array $formatters = [];
+    /** @var array<RulesetConfig> */
+    private array $rulesets = [];
+    /** @var array<string, EmitterType> */
+    private array $analyser = [];
+    /** @var array<string, array<string>> */
+    private array $skipViolations = [];
+    /** @var array<string> */
+    private array $excludeFiles = [];
+
+    public function analyser(EmitterType ...$types): self
+    {
+        foreach ($types as $type) {
+            $this->analyser[$type->value] = $type;
+        }
+
+        return $this;
+    }
+
+    public function baseline(string $baseline): self
+    {
+        /** @var array<string, array<string, array<string>>> $baselineAsArray */
+        $baselineAsArray = Yaml::parseFile($baseline);
+        /** @var array<string, string> */
+        $skipViolations = $baselineAsArray['deptrac']['skip_violations'] ?? [];
+
+        foreach ($skipViolations as $class => $skipViolation) {
+            $this->skipViolations[$class][] = $skipViolation;
+        }
+
+        return $this;
+    }
+
+    public function paths(string ...$paths): self
+    {
+        $this->paths = $paths;
+
+        return $this;
+    }
+
+    public function layer(string $name): LayerConfig
```

Comment:
Same as above. This now returns a different object, which is a bit weird to me. Maybe we could do something similar to Symfony's TreeBuilder, where we start a new definition and then have something like `end()` which will then return the related (parent) node again?

Reply:
I think a different class is not the same as returning a different instance of the same class. Here you get IDE auto-completion that is way different, helping you distinguish between these 2.